app/src/utils/data/ipfs.ts

```diff
@@ -21,7 +28,9 @@ export const uploadFile = async (file: File) => {
  * @param obj.description Description of grant
  * @returns CID
  */
-export const uploadGrantMetadata = async ({ name, description, logoURI, properties }: GrantMetadata) => {
+export const uploadGrantMetadata = async ({ name, description, logoPtr, properties }: GrantMetadata) => {
+  assertIPFSPointer(logoPtr);
+  const logoURI = getMetaPtr({ cid: (logoPtr as MetaPtr).pointer });
```

Comment:
would it make sense having `assertIPFSPointer` within `getMetaPtr` ?

Reply:
We don't want to put it in `metadataId` cause that's generic to all protocols. We could move it to `getMetaPtr` but then there's a code path in `getMetadata` where it's not called, so it seems ok to leave it as-is IMO, with the guidance being "if we're in `ipfs.ts`, as soon as we take a MetaPtr input type, validate it"

Reply:
Oh actually we can't move it to `getMetaPtr` because that only takes a CID, so by the time that method is called we've already split apart the pointer object

Reply:
Ah you're right ! Gotcha Gotcha